actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -92,10 +93,27 @@ def redirect_back(fallback_location:, allow_other_host: true, **args)
     #
     # All other options that can be passed to #redirect_to are accepted as
     # options and the behavior is identical.
-    def redirect_back_or_to(fallback_location, allow_other_host: true, **args)
-      referer = request.headers["Referer"]
-      redirect_to_referer = referer && (allow_other_host || _url_host_allowed?(referer))
-      redirect_to redirect_to_referer ? referer : fallback_location, **args
+    def redirect_back_or_to(fallback_location, allow_other_host: _allow_other_host, **options)
+      referer_location = request.referer
+      redirect_to_referer = referer_location && (allow_other_host || _url_host_allowed?(referer_location))
+      location = redirect_to_referer ? referer_location : fallback_location
+
+      allow_other_host = true if _allow_other_host && !allow_other_host # if the fallback is an open redirect
```

Comment:
Here's a potential refactor that I think could make this simpler and more readable
```suggestion
  location = request.referer || fallback_location
  location = fallback_location unless allow_other_host && _url_host_allowed?(referer_location)
  allow_other_host = false if other_hosts_forbidden
```

Reply:
Is there any way of renaming either `_allow_other_host` or `allow_other_host`? Those are similar in a way I find confusing. Seems like you could make the code clearer by choosing a more meaningful name for one of them.

Reply:
Perhaps I should elaborate on why I find this confusing.

Usually when I see a private version of a method with a matching public version, the public method is really calling the private method under the hood, and that's usually the only place the private method is called. Everywhere else, you call the public method. The public method acts as a wrapper and does some extra stuff on top, but it's basically referring to the same thing.

In this case, the two methods seem meaningfully different because it's possible for `_allow_other_host` to be true and `allow_other_host` to be false.